File: backend/src/Hotel.Orbital.Core/Models/NewsDto.cs
using System.ComponentModel.DataAnnotations;
using Entities.Enums;

namespace Core.Models;

/// <summary>
/// Dto новости
/// </summary>
public class NewsDto
{
    /// <summary>
    /// Идентфиикатор
    /// </summary>
    [Required]
    public Guid Id { get; set; }
    
    /// <summary>
    /// Город, к которому относится гостиница
    /// </summary>
    public City? City { get; set; }
    
    /// <summary>
    /// Словарь заголовков на нескольких языках
    /// </summary>
    [Required]
    public Dictionary<Language, string> Titles { get; set; }
    
    /// <summary>
    /// Словарь описаний на нескольких языках
    /// </summary>
    [Required]
    public Dictionary<Language, string> Descriptions { get; set; }
    
    /// <summary>
    /// Дата публикации
    /// </summary>
    [Required]
    public DateTimeOffset PublishedAt { get; set; }
    
    /// <summary>
    /// Обложка
    /// </summary>
    [Required]
    public ImageDto Cover { get; set; }
    
    /// <summary>
    /// Изображения
    /// </summary>
    public List<ImageDto> Images { get; set; }
}

File: admin/src/features/news/api/dto/news-dto.ts
import { fromLangRecordDto, LangRecordDto } from "@/features/misc";
import { fromImageDto, ImageDto } from "@/features/images";
import { fromNewsCityDto, NewsCityDto } from "./news-city-dto";
import { News } from "../../types/news";

export interface NewsDto {
  id: string;
  city: NewsCityDto;
  titles: LangRecordDto<string>;
  descriptions: LangRecordDto<string>;
  publishedAt: string;
  cover: ImageDto;
  images: ImageDto[];
}

export const fromNewsDto = (dto: NewsDto): News => ({
  id: dto.id,
  city: fromNewsCityDto(dto.city),
  titles: fromLangRecordDto(dto.titles),
  descriptions: fromLangRecordDto(dto.descriptions),
  publishedAt: new Date(dto.publishedAt),
  cover: fromImageDto(dto.cover),
  images: dto.images.map(fromImageDto),
});
